Userland/Libraries/LibSQL/Heap.cpp

```diff
@@ -91,11 +91,9 @@ ErrorOr<ByteBuffer> Heap::read_block(u32 block)
     TRY(seek_block(block));
 
     auto buffer = TRY(ByteBuffer::create_uninitialized(BLOCKSIZE));
-    // FIXME: This should read the entire span.
-    auto bytes = TRY(m_file->read_some(buffer));
+    TRY(m_file->read_until_filled(buffer));
 
-    dbgln_if(SQL_DEBUG, "{:hex-dump}", bytes.trim(8));
-    TRY(buffer.try_resize(bytes.size()));
+    dbgln_if(SQL_DEBUG, "{:hex-dump}", buffer.bytes().trim(8));
```

Comment:
I don't think this is right. The idea is to only read blocks (currently 1Kb in size) from the file, not the entire file as ``read_until_filled`` seems to do. There are reasons for not wanting to slurp the entire file, which may or may not be academic in the current architecture, but I suggest we don't start to mix paradigms here.